src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/utils/PsiTypeUtils.java

```diff
@@ -178,6 +185,34 @@ public static String getOptionalTypeParameter(String typeName) {
         return typeName.substring(start + 1, end);
     }
 
+    /**
+     * Returns the first type parameter for the given <code>psiType</code>.
+     * Returns the given <code>psiType</code> if the given psiType has no generic type parameter.
+     *
+     * @param psiType the psiType to return the generic type parameter for
```

Comment:
Could you add a sample  in the javadoc code like given java.util.Optinal<String> this method will return String.

Reply:
What about the impl, you did the original. Does it match the expected?